torchrl/modules/models/models.py

```diff
@@ -1032,3 +1039,164 @@ def forward(
 
         input = self.mlp(input)
         return self._lstm(input, hidden0_in, hidden1_in)
+
+
+class TanhActor(nn.Module):
+    def __init__(
+        self, out_features=None, depth=None, num_cells=None, activation_class=None, rnn_hidden_dim=200
+    ):
+        super().__init__()
+        self.backbone = MLP(
+            out_features=out_features,
+            depth=depth,
+            num_cells=num_cells,
+            activation_class=activation_class,
+        )
+        self.loc_linear = nn.Linear(out_features, out_features)
+        self.scale_linear = nn.Linear(out_features, out_features)
+        self.rnn_hidden_dim = rnn_hidden_dim
+
+    def forward(self, state, belief):
+        if belief is None:
+            *batch_size,_ = state.shape
+            belief = torch.zeros(*batch_size, self.rnn_hidden_dim, device=state.device)
+        hidden = self.backbone(state, belief)
+        loc = self.loc_linear(hidden)
+        scale = self.scale_linear(hidden)
+        if self.training:
+            action = TanhNormal(loc=loc, scale=scale).rsample()
+        else:
+            action = torch.tanh(loc)
```

Comment:
The expected value of a tanh-normal distribution is not the tanh of the mean of the normal. Let's use ProbabilisticActor with exploration_mode="mean"!